src/Authentication/Models/Challenge.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace Pinnacle\OpenIdConnect\Authentication\Models;
+
+use Pinnacle\OpenIdConnect\Support\Models\NonEmptyString;
+use Pinnacle\OpenIdConnect\Support\Traits\RandomStringGenerationTrait;
+
+class Challenge extends NonEmptyString
+{
+    use RandomStringGenerationTrait;
+
+    public static function createWithRandomChallenge(): self
+    {
+        return new self(self::generateCodeChallenge());
+    }
+
+    public function equals(Challenge $challenge): bool
```

Comment:
Typically on an `equals` function we'll allow null. That way the caller doesn't have to worry about the parameter possibly being null.